test/stdlib/tgmath.swift

```diff
@@ -103,200 +60,135 @@ let gz = CGFloat(3.3)
-  expectEqual("copysign 0.1 0.1 copysign", print3("copysign", d1, f1, g1))
-
-  (d1, f1, g1) = (nextafter(dx, dy), nextafter(fx, fy), nextafter(gx, gy))
-  expectEqual("nextafter 0.1 0.1 nextafter", print3("nextafter", d1, f1, g1))
-
-  (d1, f1, g1) = (fdim(dx, dy), fdim(fx, fy), fdim(gx, gy))
-  expectEqual("fdim 0.0 0.0 fdim", print3("fdim", d1, f1, g1))
-
-  (d1, f1, g1) = (fmax(dx, dy), fmax(fx, fy), fmax(gx, gy))
-  expectEqual("fmax 2.2 2.2 fmax", print3("fmax", d1, f1, g1))
-
-  (d1, f1, g1) = (fmin(dx, dy), fmin(fx, fy), fmin(gx, gy))
-  expectEqual("fmin 0.1 0.1 fmin", print3("fmin", d1, f1, g1))
+    expectEqualWithAccuracy(fdim(dx, dy), fdim(fx, fy), fdim(gx, gy), 0.0, ulps: 0)
+    expectEqualWithAccuracy(fmax(dx, dy), fmax(fx, fy), fmax(gx, gy), 2.2, ulps: 0)
+    expectEqualWithAccuracy(fmin(dx, dy), fmin(fx, fy), fmin(gx, gy), 0.1, ulps: 0)
 }
 
 MathTests.test("Other functions") {
 
-  (d1, d2) = modf(dy)
-  (f1, f2) = modf(fy)
-  (g1, g2) = modf(gy)
-  expectEqual("modf 2.0,0.2 2.0,0.2 modf", print6("modf", d1,d2, f1,f2, g1,g2))
-
-  (d1, f1, g1) = (ldexp(dx, ix), ldexp(fx, ix), ldexp(gx, ix))
-  expectEqual("ldexp 204.8 204.8 ldexp", print3("ldexp", d1, f1, g1))
+  expectEqual(modf(dy), (2.0, 0.20000000000000018))
+  expectEqual(modf(fy), (2.0, 0.20000005))
+#if arch(i386) || arch(arm)
+  expectEqual(modf(gy), (2.0, 0.20000005))
+#else
+  expectEqual(modf(gy), (2.0, 0.20000000000000018))
+#endif
 
-  (d1, i1) = frexp(dy)
-  (f1, i2) = frexp(fy)
-  (g1, i3) = frexp(gy)
-  expectEqual("frexp 0.55,2 0.55,2 frexp", print6("frexp", d1,i1, f1,i2, g1,i3))
+  expectEqualWithAccuracy(ldexp(dx, ix), ldexp(fx, ix), ldexp(gx, ix), 204.8, ulps: 0)
 
-  (i1, i2, i3) = (ilogb(dy), ilogb(fy), ilogb(gy))
-  expectEqual("ilogb 1 1 ilogb", print3("ilogb", i1, i2, i3))
+  expectEqual(frexp(dy), (0.55, 2))
+  expectEqual(frexp(fy), (0.55, 2))
+  expectEqual(frexp(gy), (0.55, 2))
 
-  (d1, f1, g1) = (scalbn(dx, ix), scalbn(fx, ix), scalbn(gx, ix))
-  expectEqual("scalbn 204.8 204.8 scalbn", print3("scalbn", d1, f1, g1))
+  expectEqual(ilogb(dy), 1)
+  expectEqual(ilogb(fy), 1)
+  expectEqual(ilogb(gy), 1)
 
-  (d1, i1) = lgamma(dx)
-  (f1, i2) = lgamma(fx)
-  (g1, i3) = lgamma(gx)
-  expectEqual("lgamma 2.25271265173421,1 2.25271,1 lgamma", print6("lgamma",
- d1,i1, f1,i2, g1,i3))
+  expectEqualWithAccuracy(scalbn(dx, ix), scalbn(fx, ix), scalbn(gx, ix), 204.8, ulps: 0)
 
-  (d1, i1) = remquo(dz, dy)
-  (f1, i2) = remquo(fz, fy)
-  (g1, i3) = remquo(gz, gy)
-  expectEqual("remquo 1.1,1 1.1,1 remquo", print6("remquo", d1,i1, f1,i2, g1,i3))
+  expectEqual(lgamma(dx).1, 1)
+  expectEqual(lgamma(fx).1, 1)
+  expectEqual(lgamma(gx).1, 1)
+  expectEqualWithAccuracy(lgamma(dx).0, lgamma(fx).0, lgamma(gx).0, 2.2527126517342055, ulps: 1)
 
-  (d1, f1, g1) = (nan("12345"), nan("12345"), nan("12345"))
-  expectEqual("nan nan nan nan", print3("nan", d1, f1, g1))
+  expectEqual(remquo(dz, dy).1, 1)
```

Comment:
The C stdlib `remquo` is a really weird function, and formally only guarantees that the three low-order bits of the quotient are correct. This test should test `remquo(dz, dy).1 & 7`.

Reply:
Thanks for pointing that out.  Done.